crates/pallets/starknet/src/lib.rs

```diff
@@ -731,25 +731,25 @@ pub mod pallet {
 
             match call {
                 Call::invoke { transaction } => ValidTransaction::with_tag_prefix("starknet")
-                    .priority(T::UnsignedPriority::get())
+                    .priority(u64::MAX - transaction.nonce.as_u64())
```

Comment:
comment the rationale behind this. it's not trivial to understand the reasoning and the strategy here. please explain it very clearly. eventually add examples